client/spec/test_restaurantsAPI.js

```diff
@@ -0,0 +1,534 @@
+          assert(true, 'Success function called');
+        }, 
+        function(){
+          assert(false, 'Error function called');
+        }, data);
+        this.requests[0].requestBody.should.equal(dataJson);
+        this.requests[0].url.should.equal(
+            'http://tourguide/api/orion/review/'+reviewId);
+        this.requests[0].method.should.equal( 'PATCH');
+    });
+
+
+    it('Delete a review', function() {
+      
+      var reviewId = '1';
+      
+      restaurantsAPI.deleteReview(reviewId, function(){
+          assert(true, 'Success function called');
+        }, 
+        function(){
+          assert(false, 'Error function called');
+        });
+        this.requests[0].url.should.equal(
+            'http://tourguide/api/orion/review/'+reviewId);
+        this.requests[0].method.should.equal( 'DELETE');
+    });
+
+    it('Get reviews by user', function(done) {
+      
+      var username = 'user1';
+      var dataJson = JSON.stringify(reviewsJSON);
+      
+      restaurantsAPI.getUserReviews(username, function(result){
+          assert(true, 'Success function called');
+          JSON.parse(result).should.deep.equal(reviewsJSON);
+          done();
+        }, 
+        function(){
+          assert(false, 'Error function called');
+          done();
+        });
+        this.requests[0].url.should.equal(
+            'http://tourguide/api/orion/reviews/user/'+username);
+        this.requests[0].method.should.equal( 'GET');
+        this.requests[0].respond(200, { 'Content-Type': 'text/json' }, dataJson);
+    });
+
+    it('Get reviews by restaurant', function(done) {
+      
+      var restaurantName = 'restaurant1';
+      var dataJson = JSON.stringify(reviewsJSON);
+      
+      restaurantsAPI.getRestaurantReviews(restaurantName, function(result){
+          assert(true, 'Success function called');
+          JSON.parse(result).should.deep.equal(reviewsJSON);
+          done();
+        }, 
+        function(){
+          assert(false, 'Error function called');
+          done();
+        });
+        this.requests[0].url.should.equal(
+            'http://tourguide/api/orion/reviews/restaurant/'+restaurantName);
+        this.requests[0].method.should.equal( 'GET');
+        this.requests[0].respond(200, { 'Content-Type': 'text/json' }, dataJson);
+    });
+
+    it('Get reviews by organization', function(done) {
+      
+      var organizationName = 'organization1';
+      var dataJson = JSON.stringify(reviewsJSON);
+      
+      restaurantsAPI.getOrganizationReviews(organizationName, function(result){
+          assert(true, 'Success function called');
+          JSON.parse(result).should.deep.equal(reviewsJSON);
+          done();
+        }, 
+        function(){
+          assert(false, 'Error function called');
+          done();
+        });
+        this.requests[0].url.should.equal(
+            'http://tourguide/api/orion/reviews/organization/'+organizationName);
```

Comment:
instead of repiting http://tourguide/api/... etc., why don't you create a constant, at least for the base URL?